design/one-pager-provider-nop..md

```diff
@@ -0,0 +1,31 @@
+# Crossplane provider-nop
+
+* Owner: Rahul Grover (@rahulgrover99)
+* Reviewers: Crossplane Maintainers
+* Status: Speculative
+
+## Introduction
+
+There is a lot of machinery around the composition engine that needs to be tested which raises demand for a provider on which could rely to behave the way we want it to. It can be used to mock the behavior of providers by creating fake objects whenever needed so that we could test the response of composition to the resource becoming ready or not. 
+
+This provider does exactly what you tell it to. It can be used for testing functionality in core Crossplane without needing to account for the requirements of external APIs. It supports a single managed resource that contains a spec that allows for specifying how and when the status of the resource is changed. It does not orchestrate any external system. 
+
+This provider will be installed while running Crossplane end-to-end tests to satisfy any kind of composite infrastructure resource by doing nothing to simulate different scenarios. 
+
+## Design
+
+The provider will have a single managed resource `NopResource` that will reflect whatever is in the spec into the status. It will let you define to:
+- become ready after this period
+- become unhealthy after this period
+
+The config might look something like:
+
+```yaml
+spec:
+    condtionAfter:
+        time: 5s
+        condition: ready
+```
```

Comment:
another field in the spec that I can imagine would be a receiver field to test that patches make it into the nop object when it's part of a composition.  e.g.:

* nop object spec has a field called 'patchReceiverField` of type string
* XRD has a field called `coolField` of type string
* The composition under the XRD creates 1 nop object and patches the value from `coolField` onto this receiver field on the nop object.
* test case verifies that the value of `coolField` made it successfully into the nop object's receiver field